erpnext/accounts/doctype/auto_reconcile/auto_reconcile.py

```diff
@@ -0,0 +1,407 @@
+from frappe.utils import get_link_to_form
+from frappe.utils.scheduler import is_scheduler_inactive
+
+
+class AutoReconcile(Document):
+	def validate(self):
+		self.validate_receivable_payable_account()
+		self.validate_bank_cash_account()
+
+	def validate_receivable_payable_account(self):
+		if self.receivable_payable_account:
+			if not frappe.db.get_list(
+				"Account", filters={"name": self.receivable_payable_account, "company": self.company}
+			):
+				frappe.throw(
+					_("Receivable/Payable Account: {0} doesn't below to company {1}").format(
+						frappe.bold(self.receivable_payable_account), frappe.bold(self.company)
+					)
+				)
+
+	def validate_bank_cash_account(self):
+		if self.bank_cash_account:
+			if not frappe.db.get_list(
+				"Account", filters={"name": self.bank_cash_account, "company": self.company}
+			):
+				frappe.throw(
+					_("Bank/Cash Account {0} doesn't below to company {1}").format(
+						frappe.bold(self.bank_cash_account), frappe.bold(self.company)
+					)
+				)
+
+	def before_save(self):
+		self.status = "Draft"
+		self.error_log = ""
+
+	def before_submit(self):
+		self.status = "Queued"
+		self.error_log = ""
+
+
+@frappe.whitelist()
+def get_progress(docname: str | None = None) -> float:
+	progress = 0.0
+	if docname:
+		reconcile_log = frappe.db.get_list(
+			"Auto Reconcile Log", filters={"auto_reconcile": docname}, as_list=1
+		)
+		if reconcile_log:
+			processed = frappe.db.get_value("Auto Reconcile Log", reconcile_log[0][0], "reconciled_entries")
+			total = frappe.db.get_value("Auto Reconcile Log", reconcile_log[0][0], "total_allocations")
+
+			if processed != 0:
+				progress = processed / total * 100
+			elif total == 0:
+				progress = 100
+
+			return progress
+
+	return progress
+
+
+def get_pr_instance(doc: str):
+	auto_reconcile = frappe.get_doc("Auto Reconcile", doc)
+
+	pr = frappe.get_doc("Payment Reconciliation")
+	fields = [
+		"company",
+		"party_type",
+		"party",
+		"receivable_payable_account",
+		"from_invoice_date",
+		"to_invoice_date",
+		"from_payment_date",
+		"to_payment_date",
+	]
+	d = {}
+	for field in fields:
+		d[field] = auto_reconcile.get(field)
+	pr.update(d)
+	return pr
+
+
+def is_job_running(job_name: str) -> bool:
+	jobs = frappe.db.get_list("RQ Job", filters={"status": ["in", ["started", "queued"]]})
+	for x in jobs:
+		if x.job_name == job_name:
+			return True
+	return False
+
+
+@frappe.whitelist()
+def run_reconciliation_job():
+	"""
+	Trigger background job for doc if no other doc is running
+	"""
+	if not frappe.db.get_single_value("Accounts Settings", "enable_auto_reconciliation"):
+		frappe.throw(
+			_("Auto Reconciliation has been disabled. Enable it through {}").format(
+				get_link_to_form("Accounts Settings", "Accounts Settings")
+			)
+		)
+
+		return
+
+	if not is_scheduler_inactive():
+		res = is_any_doc_running()
+		if not res:
+			# Get Queued item and start the process
+			queued = frappe.db.get_list(
```

Comment:
`get_all` is preferred on server side, `get_list` does permission checks and can cause unwanted errors in some cases

Reply:
replaced with `get_all` as suggested

Reply:
addressed